src/encoded/static/components/search.js

```diff
@@ -1903,51 +1902,48 @@ export const SearchControls = ({ context, visualizeDisabledTitle, showResultsTog
         );
 
     let resultsToggle = null;
+    const parsedUrl = React.useMemo(() => url.parse(context['@id']), [context]);
+    const query = React.useMemo(() => new QueryString(parsedUrl.query), [parsedUrl]);
+    // Get the current value of the "limit=x" query string parameter. No "limit=x" means the
+    // default value applies. The back end allows exactly zero or one "limit=x" parameter.
+    const pageLimit = React.useMemo(() => {
+        const limitValues = query.getKeyValues('limit');
+        return limitValues.length === 1 ? Number(limitValues[0]) || DEFAULT_PAGE_LIMIT : DEFAULT_PAGE_LIMIT;
+    }, [query]);
+    const pageLimitOptions = [25, 50, 100, 200];
+
     if (showResultsToggle) {
```

Comment:
I also tried including the condition (context.total > results.length) so in that case if you filter down to 5 results, the box of results limits doesn't display at all. I decided that I didn't like having the results limits appear and disappear depending on selections. Similarly, I thought about not displaying the "200" if there were say 95 results total (having "100" be the max in that case), but thought it was better to just keep all the options for all cases. I can see the case for the other solution though.

Reply:
Yeah, seems reasonable to keep the controls there. If ENCODE was still in active development I might display it disabled, but this is probably fine as is. I doubt you’re getting any advantage from caching the query with `useMemo`. `context` only changes on user command, and the calculation for `parsedUrl` and `query` is pretty simple to do on every render. The caches will just take up memory, though they’re not very big either.